tendrl/commons/flows/unmanage_cluster/__init__.py

```diff
@@ -0,0 +1,72 @@
+import etcd
+import time
+
+from tendrl.commons import flows
+from tendrl.commons.flows.exceptions import FlowExecutionFailedError
+from tendrl.commons.objects import AtomExecutionFailedError
+
+
+class UnmanageCluster(flows.BaseFlow):
+    def __init__(self, *args, **kwargs):
+        super(UnmanageCluster, self).__init__(*args, **kwargs)
+
+    def run(self):
+        integration_id = self.parameters['TendrlContext.integration_id']
+        _cluster = NS.tendrl.objects.Cluster(
+            integration_id=integration_id
+        ).load()
+        if _cluster.status is not None and \
+            _cluster.status != "" and \
+            _cluster.status in \
+            ['syncing', 'importing', 'unmanaging']:
+            raise FlowExecutionFailedError(
+                "Another job in progress for cluster"
+                "please wait till the job finishes "
+                "(job_id: %s) (integration_id: %s) " %
+                (
+                    _cluster.current_job['job_id'],
+                    _cluster.integration_id
+                )
+            )
+
+        _cluster.status = "unmanaging"
+        _cluster.current_job = {
+            'job_id': self.job_id,
+            'job_name': self.__class__.__name__,
+            'status': "in_progress"
+        }
+        _cluster.save()
+
+        try:
+            super(UnmanageCluster, self).run()
+            # Wait for cluster to re-appear as detected cluster
+            # as node-agents are still running on the storage nodes
+            while True:
+                try:
+                    _cluster = NS.tendrl.objects.Cluster(
+                        integration_id=integration_id
+                    ).load()
+                    if _cluster.is_managed == "no":
+                        break
+                except etcd.EtcdKeyNotFound:
+                    time.sleep(5)
+                    continue
+            _cluster.status = ""
+            _cluster.current_job['status'] = "done"
```

Comment:
we have been using "finished" all over Tendrl, lets stick to that instead of "done"

Reply:
done